app/models/notification.rb

```diff
@@ -0,0 +1,31 @@
+class Notification < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+  belongs_to :user
+  belongs_to :course
+
+  scope :center_popup, -> { where(type: 'Notification::CenterPopup') }
+  scope :right_side_popup, -> { where(type: 'Notification::RightSidePopup') }
+  scope :email, -> { where(type: 'Notification::EmailNotification') }
+
+  def self.notify(user, course, options = {})
+    case options[:type]
```

Comment:
The notification is the same. It is the mechanism by which the user sees the notification that is different. The API to create a new notification is the only way the programmer will create them. Presentation is handled by the framework.